File: cs/Wpf-Dashboard-Custom-Properties/Modules/ChartItemModule/ChartOptions/ScaleBreak.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Windows.Data;
using DevExpress.DashboardCommon;
using DevExpress.DashboardWpf;
using DevExpress.Xpf.Charts;
using DevExpress.XtraReports.UI;

namespace Wpf_Dashboard_Custom_Properties {
    public class ScaleBreaksConverter : IMultiValueConverter {
        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture) {
            string itemName = (string)values[0];
            IDashboardControlProvider provider = (IDashboardControlProvider)values[1];
            ChartDashboardItem chartItem = provider.Dashboard.Items[itemName] as ChartDashboardItem;
            bool scaleBreakEnabled = System.Convert.ToBoolean(chartItem.CustomProperties[ChartItemModule.ScaleBreakPropertyName]);
            return new AutoScaleBreaks() { Enabled = scaleBreakEnabled };
        }
        public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }

    public static class ScaleBreakExtensions {
        public static void CustomExport(DashboardControl dashboardControl, CustomExportEventArgs e) {
            Dictionary<string, XRControl> controls = e.GetPrintableControls();
            foreach(var control in controls) {
                XRChart xrChart = control.Value as XRChart;
                if(xrChart != null && xrChart.Diagram != null) {
                    DashboardItem chartItem = dashboardControl.Dashboard.Items[control.Key];
                    bool scaleBreakEnabled = Convert.ToBoolean(chartItem.CustomProperties[ChartItemModule.ScaleBreakPropertyName]);
                    ((DevExpress.XtraCharts.XYDiagram)xrChart.Diagram).SecondaryAxesY[0].AutoScaleBreaks.Enabled = scaleBreakEnabled;
                }
            }
        }
    }
}

File: VB/Wpf-Dashboard-Custom-Properties/Modules/ChartItemModule/ChartOptions/ScaleBreak.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Windows.Data
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWpf
Imports DevExpress.Xpf.Charts
Imports DevExpress.XtraReports.UI

Namespace Wpf_Dashboard_Custom_Properties
	Public Class ScaleBreaksConverter
		Implements IMultiValueConverter
		Public Function Convert(ByVal values() As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
			Dim itemName As String = CStr(values(0))
			Dim provider As IDashboardControlProvider = CType(values(1), IDashboardControlProvider)
			Dim chartItem As ChartDashboardItem = TryCast(provider.Dashboard.Items(itemName), ChartDashboardItem)
			Dim scaleBreakEnabled As Boolean = System.Convert.ToBoolean(chartItem.CustomProperties(ChartItemModule.ScaleBreakPropertyName))
			Return New AutoScaleBreaks() With {.Enabled = scaleBreakEnabled}
		End Function
		Public Function ConvertBack(ByVal value As Object, ByVal targetTypes() As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class

	Public NotInheritable Class ScaleBreakExtensions
		Private Sub New()
		End Sub
		Public Shared Sub CustomExport(ByVal dashboardControl As DashboardControl, ByVal e As CustomExportEventArgs)
			Dim controls As Dictionary(Of String, XRControl) = e.GetPrintableControls()
			For Each control In controls
				Dim xrChart As XRChart = TryCast(control.Value, XRChart)
				If xrChart IsNot Nothing AndAlso xrChart.Diagram IsNot Nothing Then
					Dim chartItem As DashboardItem = dashboardControl.Dashboard.Items(control.Key)
					Dim scaleBreakEnabled As Boolean = Convert.ToBoolean(chartItem.CustomProperties(ChartItemModule.ScaleBreakPropertyName))
					CType(xrChart.Diagram, DevExpress.XtraCharts.XYDiagram).SecondaryAxesY(0).AutoScaleBreaks.Enabled = scaleBreakEnabled
				End If
			Next control
		End Sub
	End Class
End Namespace
